src/editor/Collaboration.js

```diff
@@ -125,6 +139,35 @@ define(function (require, exports, module) {
             }
         }
         _changing = false;
+        peer.on("fileTransfer", function (metadata, receiver) {
+            console.log("incoming filetransfer", metadata.name, metadata);
+            receiver.on("progress", function (bytesReceived) {
+                //TODO:: Add UI element to show percentage of file received.
+                console.log("Percentage of file received : " + (bytesReceived / metadata.size * 100));
+            });
+            receiver.on("receivedFile", function (file, metadata) {
+                receiver.channel.close();
+                var reader = new window.FileReader();
+                reader.onload = function(e) {
+                    var data = e.target.result;
+                    var buffer = new FilerUtils.Buffer(data);
+                    var filename = Path.join(StartupState.project("root"), metadata.name);
+                    //keep this file in the received array to make sure we don't emit back this file.
+                    _received[metadata.name] = true;
+                    FilerUtils.writeFileAsBinary(filename, buffer, function(err) {
+                        if(err) {
+                            // TODO :: Ask the user for the file again.
+                            console.log(err);
+                        };
+                        // Clear _received of file name for future events.
+                        window.setTimeout(function() {
+                            delete _received[metadata.name];
```

Comment:
Your thoughts on the timeout @gideonthomas . The callback gets called before the trigger of [this](https://github.com/mozilla/brackets/pull/855/files#diff-172c2aa71bb7326386a85125ccd13603R60) event. To make sure the received object has a reference to this fie path, I added a timeout. 
Did something similar for renaming files [here](https://github.com/mozilla/brackets/pull/869/files#diff-172c2aa71bb7326386a85125ccd13603R97)